pages/linux/udevadm.md

```diff
@@ -0,0 +1,28 @@
+# udevadm
+
+> Linux `udev` management tool.
+> More information: <https://www.freedesktop.org/software/systemd/man/udevadm>.
+
+- Monitor all device events:
+
+`sudo udevadm monitor`
+
+- Print `uevents` sent out by the kernel:
+
+`sudo udevadm monitor --kernel`
+
+- Print device events after rule processing by `udev`:
+
+`sudo udevmadm monitor --udev`
```

Comment:
```suggestion
- Print device events after rules processed by `udev`:

`sudo udevadm monitor --udev`
```

Reply:
I don't see a problem with `after rule processing`. But maybe we could still rephrase it. What are those rules for? Do they apply on the devices or on the events? Is ``Print device events after being filtered by `udev` `` correct?

Reply:
I don't see a problem with it either, but I'm happy to take suggestions.

The rules are meant to decide how the device is interfaced with in user-space. You create a rule like this and add it to `/etc/udev/rules.d/`:
```
SUBSYSTEMS=="usb", ATTRS{idVendor}=="1050", ATTRS{idProduct}=="0407", SYMLINK+="/dev/myDevice"
```
Then you plug a device in, the kernel passes up the `uevents`, and `udev` processes them. If they match (`==`), it runs an action. In this case, symlinking to `/dev/myDevice`. Passing in the `--udev` rule, will print any matching rules that **were** processed.

Reply:
Some suggestions:
> Print device events and all matched `udev` rules

> Print device events along with all matched `udev` rules

> Print device events and `udev` rules in case one matched

I hope I understand this correctly. Or does it only print device events which had a matchnig rule?

Reply:
It doesn't print the `udev` rule specifically. It prints the device events that `udev` sends out after processing the rule. Here's an example of me unplugging a USB drive:
```
$ sudo udevadm monitor -u
monitor will print the received events for:
UDEV - the event which udev sends out after rule processing

UDEV  [156502.669688] remove   /devices/pci0000:00/0000:00:1d.0/0000:03:00.0/0000:04:01.0/0000:06:00.0/0000:07:01.0/0000:09:00.0/usb7/7-2/7-2.1/7-2.1:1.0
```

Reply:
Then I think we can either keep the original or change it to 
> Print device events after being processed by `udev`.

Reply:
Sounds good -- I'll go ahead and use your suggestion